### PR TITLE
Feature/save user data

### PR DESCRIPTION
**Pull request by Erbidium**

### CODE REVIEW

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs [hunk 1]
```diff
@@ -48,6 +48,21 @@ private async Task<User> GetUserById(long id)
...
 
+        public async Task<UserDto> CreateUserPreferences(NewUserDto userDto)
+        {
+            var newUser = _mapper.Map<NewUserDto, User>(userDto);
+
+            var existingUser = await _context.Users.FirstOrDefaultAsync(u => u.Email == userDto.Email);
+            if (existingUser is not null)
+            {
+                return _mapper.Map<User, UserDto>(existingUser);
+            }
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: I believe if we create a user we can omit this check
And just throw an error that user with such email already exists

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -9,6 +9,7 @@
 using Microsoft.AspNetCore.Authentication.JwtBearer;
 using Microsoft.IdentityModel.Tokens;
 using Azure.Storage.Blobs;
+using EasyMeets.Core.WebAPI.Validators.Sample;
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs]: Why did you add this?

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿using EasyMeets.Core.Common.DTO;
 using FluentValidation;
 
-namespace EasyMeets.Core.WebAPI.Validators
+namespace EasyMeets.Core.WebAPI.Validators.Sample
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs]: Why did you add this?

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts [hunk 4]
```diff
@@ -15,6 +22,8 @@ export class SignUpFormComponent {
 
     public hidePassword = true;
 
+    private unsubscribe$ = new Subject<void>();
+
```
Comment (tatianahutii) [inline frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: Why did you add this?

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts [hunk 5]
```diff
@@ -33,18 +42,44 @@ export class SignUpFormComponent {
...
     });
 
-    // eslint-disable-next-line no-empty-function
-    constructor(private authService: AuthService, private router: Router) {}
+    constructor(
+        private authService: AuthService,
+        private router: Router,
+        private userService: UserService,
+        private notifications: NotificationService,
+        // eslint-disable-next-line no-empty-function
```
Comment (tatianahutii) [inline frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: Please remove this

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts [hunk 6]
```diff
@@ -62,4 +97,35 @@ export class SignUpFormComponent {
...
+            userLanguage = Language.Ukr;
+        } else if (userLanguageBrowser.includes('pl')) {
+            userLanguage = Language.Pl;
+        } else if (userLanguageBrowser.includes('fr')) {
+            userLanguage = Language.Fr;
+        } else if (userLanguageBrowser.includes('it')) {
+            userLanguage = Language.It;
+        } else {
+            userLanguage = Language.Eng;
+        }
```
Comment (tatianahutii) [inline frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: please move this to a separate method (helper) and use switch/case

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs (hunk 3 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs]: Fixed namespace in order to correspond to file location

Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs]: Sample validatior is in the separate folder, but namespace was Validators, not Validators.Sample

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts (hunk 4 above)
Comment (Erbidium) [reply frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: I use subscription on response and then i show pop up notification. I thought it is necessary in such cases

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts (hunk 5 above)
Comment (Erbidium) [reply frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: But it's an empty constructor with injected services

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs (hunk 1 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: If user registers via google again, he will see this error. This is the reason why I have added this check. I just remember that on different websites if user try to register again with google, he will see no error and just logins

Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: So the google user doesn't see any error, but normal user will see the error because he will get an error from the firebase

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs]: Fixed namespace in order to correspond to file location

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts (hunk 6 above)
Comment (Erbidium) [reply frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: Done

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts (hunk 4 above)
Comment (tatianahutii) [reply frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: Yes, but you need to use this from BaseComponent

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs (hunk 3 above)
Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs]: It would be better to have one namespace for all validators instead to have a separate namespace for everyone

Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs]: Okay, I will fix it soon

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts (hunk 5 above)
Comment (tatianahutii) [reply frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: You need to extend BaseComponent and use super() here

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts (hunk 4 above)
Comment (Erbidium) [reply frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: Done

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs (hunk 3 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Validators/Sample/NewSampleDtoValidator.cs]: Fixed

File: backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs]: Fixed

File: frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts (hunk 5 above)
Comment (Erbidium) [reply frontend/src/app/modules/auth/components/sign-up-form/sign-up-form.component.ts]: Fixed

File: backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs (hunk 1 above)
Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: So, should I remove it or leave?

Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: for now, we don't need this, and for now we will think that this is two different users

Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: Okay, start fixing)

Comment (Erbidium) [reply backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/UserService.cs]: Fixed

File: frontend/src/app/core/helpers/language-helper.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            text: 'it',
+            enum: Language.It,
+        },
+    ];
+
+    languages.forEach(lang => {
+        if (userLanguageBrowser.includes(lang.text)) {
+            userLanguage = lang.enum;
+        }
+    });
```
Comment (tatianahutii) [inline frontend/src/app/core/helpers/language-helper.ts]: what is this part supposed to do?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/core/helpers/language-helper.ts (hunk 7 above)
Comment (Erbidium) [reply frontend/src/app/core/helpers/language-helper.ts]: It will iterate through array with pairs of languages enums and texts. If language of the user contains part of the text, the corresponding enum will be selected. We can have "en" or "en-us" from browser api and these two are english languages, so I look for substrings of languages

Comment (Erbidium) [reply frontend/src/app/core/helpers/language-helper.ts]: Refactored helper)

**Review by tatianahutii — APPROVED** [review]
LGTM
